tests/unittests/EntryPoint.cpp

```diff
@@ -15,21 +15,27 @@
 //******************************************************************************
 
 #include <windows.h>
-
 #include <TestFramework.h>
 
 int main(int argc, char** argv) {
+// ::CoInitialize, Uninitialize, don't exist on OSX, aren't needed
+#ifdef WIN32
     // Initialize COM for all of the tests
     ::CoInitializeEx(NULL, COINIT_MULTITHREADED);
 
     testing::InitGoogleTest(&argc, argv);
-
     LOG_INFO("Starting unit tests...");
-
     auto result = RUN_ALL_TESTS();
 
     // TODO: move to a scopeguard when we have one in our codebase
     ::CoUninitialize();
 
+#else
+    testing::InitGoogleTest(&argc, argv);
+    printf("Starting unit tests...\n");
+    auto result = RUN_ALL_TESTS();
```

Comment:
Should be pretty easy to make the change and then only scope guard is if def'ed. The idea is to use an RAII type to contol the CoInit / uninit calls so that if an exception occurs, we still uninit properly. http://stackoverflow.com/questions/31365013/what-is-scopeguard-in-c for the general concept in question. We can talk in person about if we already have one implemented somewhere.